ui/src/app/certificate_requests/components.tsx

```diff
@@ -28,6 +31,95 @@ export function ConfirmationModal({ modalData, setModalData }: ConfirmationModal
     )
 }
 
+function SubmitCertificate({ existingCSRText, existingCertText, certText, onClickFunc }: { existingCSRText: string, existingCertText: string, certText: string, onClickFunc: any }) {
+    let certIsValid = false
+    try {
+        extractCert(certText.trim())
+        if (csrMatchesCertificate(existingCSRText.trim(), certText.trim())) {
+            certIsValid = true
+        }
+    }
+    catch { }
```

Comment:
Here we see "Invalid Certificates" when the certificate is valid but does not match the CSR. I believe we should differentiate those 2 situations with independent error messages.
- The Certificate is invalid
- The Certificate does not match the request

Reply:
done